### PR TITLE
Assignment 4

### LINKED ISSUE #118: Basic authentication
Using the `hapi-auth-basic` plugin, add a new `/private` endpoint protected by a username and password that returns a simple HTML page with a welcome message. The welcome message should include the logged in username. Do not use any templates (views) at this stage. For username/password validation, create a static `lib/users.json` file (we will replace this in the near future) that contains a few sample users and passwords.

The new endpoint will be contained in its own plugin, similar to how we set up `/version`, and will require bringing test coverage back to 100%.

Due: 4/27

### PR DESCRIPTION
**Pull request by idanwe**
* Add /private route
 * Add hapi-auth-basic
 * Rename Server to Huenversity
 * Closes #118

### CODE REVIEW

File: lib/private.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+                    var username = request.auth.credentials.username;
+                    return reply(internals.respone(username));
+                }
+            }
+        });
+
+        return next();
+    });
+
```
Comment (AdriVanHoudt) [inline lib/private.js]: no need for newline here

File: lib/private.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+};
+
+exports.register.attributes = {
+    name: 'private'
+};
+
+internals.validate = function (username, password, callback) {
```
Comment (AdriVanHoudt) [inline lib/private.js]: personally I prefer declaring functions before usage

File: test/private.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            var request = {
+                url: '/private',
+                headers: {
+                    authorization: internals.header('user1', 'secret1')
+                }
+            };
+            server.inject(request, function (res) {
+
+                expect(res.statusCode).to.equal(200);
+                expect(res.result).to.contains('user1');
```
Comment (AdriVanHoudt) [inline test/private.js]: matching the full res.result object would tighten up the test, imagine the server adding an extra key, this test will not catch that

File: lib/private.js (hunk 2 above)
Comment (idanwe) [inline lib/private.js]: I think it is ok, for example from the [hoek code](https://github.com/hapijs/hoek/blob/master/lib/index.js) 

And I find it more clean

Comment (AdriVanHoudt) [inline lib/private.js]: yeah true, probably because of my Java background haha

**Comment by FennNaten** [comment]
Hi,
I can see that you didn't updated the package.json to add hapi-auth-basic module. Maybe you forgot to install it with --save flag.

File: lib/private.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        server.route({
+            method: 'GET',
+            path: '/private',
+            config: {
+                description: 'Returns welcome message with the authenticated username',
+                auth: 'simple',
+                handler: function (request, reply) {
+
+                    var username = request.auth.credentials.username;
+                    return reply(internals.respone(username));
```
Comment (FennNaten) [inline lib/private.js]: little typo here, you typed 'respone' instead of 'response'

File: lib/private.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    var user = UsersData[username];
+    if (user && user.password === password) {
+        return callback(null, true, user);
+    }
+
+    return callback(null, false);
+};
+
+internals.respone = function (username) {
```
Comment (FennNaten) [inline lib/private.js]: same typo as above ;)

Comment (idanwe) [inline lib/private.js]: @FennNaten thanks

**Pull request closed without merging**